runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -358,19 +376,19 @@ public void insert(@ConflictParameter K key, V value) {
      */
     @Accessor
     public List<V> scanAndFilter(Predicate<? super V> valuePredicate) {
-        return mainMap.entryStream()
+        return pool.submit(() -> mainMap.entryStream()
```

Comment:
scanAndFilter/scanAndFilterByEntry are not the only functions using entryStream(), which uses parallel stream. Other functions like keySet() are using it as well, why only scanAndFilter/scanAndFilterByEntry are changed in the patch?

Reply:
I didn't notice.

I'm going to change them to use keySet and values instead, because those two methods shouldn't be exposed by `PersistedStreamingMap`